### PR TITLE
feat: Allow setting SDK info with Options initWithDict

### PR DESCRIPTION
**Pull request by vaind**
## :scroll: Description

Enables users to override the SDK info

## :bulb: Motivation and Context

This allows other SDKs that use sentry-cocoa internally to identify the events. Enables https://github.com/getsentry/sentry-unity/issues/616

## :green_heart: How did you test it?

Added a test case

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryOptions.m [hunk 1]
```diff
@@ -279,6 +279,10 @@ - (BOOL)validateOptions:(NSDictionary<NSString *, id> *)options
 
     [self setBool:options[@"sendClientReports"]
             block:^(BOOL value) { self->_sendClientReports = value; }];
+    
+    if ([options[@"sdk"] isKindOfClass:[NSDictionary class]]) {
```
Comment (philipphofmann) [inline Sources/Sentry/SentryOptions.m:283]: `h`: I think this should be `sdkInfo`:
```suggestion
    if ([options[@"sdkInfo"] isKindOfClass:[NSDictionary class]]) {
```

**Review by philipphofmann — COMMENTED** [review]
One issue to address. Thanks for doing this 🚀

File: Sources/Sentry/SentryOptions.m (hunk 1 above)
Comment (vaind) [reply Sources/Sentry/SentryOptions.m:283]: I agree and would have called it that (and passed the directory content, not the whole options), but the SdkInfo already expect this dictionary structure, including the "sdk": https://github.com/getsentry/sentry-cocoa/blob/d68e2b6e0e62b0e26fede487eb5c2642825fa3fe/Sources/Sentry/SentrySdkInfo.m#L34

So would you prefer the options would look like this?
```
options: {
  sdkInfo: {
    sdk: {
      name: ...
      version
    }  
  }
}
```

Comment (philipphofmann) [reply Sources/Sentry/SentryOptions.m:283]: Sorry I missed that. Then maybe worth a comment

Comment (vaind) [reply Sources/Sentry/SentryOptions.m:285]: added the comment

**Review by brustolin — APPROVED** [review]
LGTM

**Comment by imanshahpoury** [comment]
Pishro🤩

**Pull request merged**